### ISSUE TITLE
Edit button does not show up on Hover Menu for recently sent message

### ISSUE THREAD

**Issue by isagoico** [Expensify/Expensify.cash#2966]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Expected Result:
User should be able to edit the message even immediately after sending

## Actual Result:
User is unable to edit the message. When navigating away and back to the conversation the option is displayed. 

## Action Performed:
1. Open desktop app
2. Navigate to a conversation
3. Send a message
4. Right click on the message that was just sent

## Workaround:
Navigating to another conversation and the back solves the issue. 

## Platform:
Where is this issue occurring?

 Web ✔️ 
 iOS
 Android
 Desktop App ✔️ 
 Mobile Web

**Version Number:** 1.0.47-0

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** 

![image](https://user-images.githubusercontent.com/44479856/118547260-1a6eab00-b727-11eb-822c-57647e2208db.png)

**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

___
From @puneetlath https://expensify.slack.com/archives/C01GTK53T8Q/p1621262754223000
> it doesn’t seem that I’m able to edit the message I just sent right after I send it. The edit icon doesn’t show up. Only if I click out to another chat and then click back to the one I’m on do I see the edit option (on mac desktop)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @SofiedeVreese (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by SofiedeVreese** [comment]
Yup can reproduce steps:

https://user-images.githubusercontent.com/40040992/118607645-1ab78680-b7b9-11eb-8ffc-58442aed4349.mov

Getting an engineer to look at this- and decide whether it's a good one for an external contributor.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @NikkiWines (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by NikkiWines** [comment]
Also able to reproduce - this seems fine for Upwork but cc'ing @yuwenmemon since this looks related to your [`Edit Comments` PR ](https://github.com/Expensify/Expensify.cash/pull/2320), not sure if you want to handle this internally or not.

**Comment by parasharrajat** [comment]
Aha, Seems we are not refreshing the contextMenu Items as they are initialized on the constructor.

---
There is a catch here. we need `actionId` from the backend before we can really edit it. So It takes a sec or two before showing the edit icon. This is the same for both Right-click context menu and Hover Menu.

 *but for this issue, Icon is not shown at all in the hover menu.*
---

A couple of things could be done here to fix it.
1. Either move contextMenu initialization to render method.
https://github.com/Expensify/Expensify.cash/blob/a87db63bcd2fc9b5fe8c70b1f7878102721c2700/src/pages/home/report/ReportActionContextMenu.js#L67
2. Update `contextMenu` on props changes on `componentDidUpdate`. It does not Update the menu, if already visible. We need to hide the menu and then hover again to see it.
3. Convert `shouldshow` to function (Best). It will show the Icon even if you hover menu is already visible.

**Comment by yuwenmemon** [comment]
I think there might be a larger discussion to be had for this one. Currently, as @parasharrajat observed, with edit messages, we can only edit a message once it's been posted to the DB (and thus has a `reportActionID`). 

> Convert `shouldshow` to function (Best). It will show the Icon even if you hover menu is already visible.

What does this mean exactly? Why do you think it's the best solution?

**Comment by parasharrajat** [comment]
I meant this 
https://github.com/Expensify/Expensify.cash/blob/a87db63bcd2fc9b5fe8c70b1f7878102721c2700/src/pages/home/report/ReportActionContextMenu.js#L115

TO
```
    shouldShow: () => this.props.reportAction.actorEmail === this.props.session.email
                    && !isReportMessageAttachment(this.getActionText())
                    && this.props.reportAction.reportActionID,

```

Why this is best?

We don't need to update the whole context menu ourselves after checking the prop changes. Instead on each render, this function will be called and the changes would be in real-time. Even if the menu is already visible, It's kind of cheap.

**Comment by puneetlath** [comment]
>I think there might be a larger discussion to be had for this one. Currently, as @parasharrajat observed, with edit messages, we can only edit a message once it's been posted to the DB (and thus has a reportActionID).

Yes, I agree. We should discuss this in slack. I think the most common time people will want to edit is immediately after they hit send. So even a second delay is an annoyance.

**Comment by puneetlath** [comment]
Well I think there are two things here. 

1. Bug: the option to edit the comment never appears in the hover menu after sending a message unless you click out of the chat and back into it
2. Improvement: the ability to edit a comment immediately after hitting send, without needing to wait until we get the `actionID` back from the server

**Comment by thienlnam** [comment]
Ah my bad missed that - updated the title again to reflect that

**Comment by roryabraham** [comment]
I think `2` will have to be internal, and `1` will be solved [over here](https://github.com/Expensify/Expensify.cash/pull/2046)

**Comment by NikkiWines** [comment]
Sounds like we might not need this issue then? Should we close this out?

**Comment by yuwenmemon** [comment]
I think we should fix `1` first before evaluating if `2` is indeed a problem. From what I've been seeing the reportActionID coming back from the server is pretty instantaneous in that you can edit a comment without much delay right now as long as you right-click. So it might not be as big of a deal as we think. Observe:
![Kapture 2021-05-21 at 11 21 21](https://user-images.githubusercontent.com/4741899/119181676-c6b1d980-ba26-11eb-9742-cf573adb24a4.gif)

But yes, if the bug of that option not showing in the hover menu will be solved in the Delete Comments PR, we can close this be closed out by that. Since that's what the OP is trying to address.

**Comment by puneetlath** [comment]
Yep, sounds good. Let's just focus on that for now. And given that's being handled in another PR, I'll go ahead and close this.
